lib/ladybug/energy_model/schedule_ruleset.rb

```diff
@@ -0,0 +1,134 @@
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'ladybug/energy_model/model_object'
+
+require 'json-schema'
+require 'json'
+require 'openstudio'
+
+module Ladybug
+  module EnergyModel
+    class ScheduleRuleset < ModelObject
+      attr_reader :errors, :warnings
+            
+      def initialize(hash = {})
+        super(hash)
+      end
+
+      def defaults
+        result = {}
+        result
+      end
+
+      def find_existing_openstudio_object(openstudio_model)
+        object = openstudio_model.getScheduleRulesetByName(@hash[:name])
+        return object.get if object.is_initialized
+        nil
+      end
+
+
+      def create_openstudio_object(openstudio_model)
+        openstudio_schedule_ruleset = OpenStudio::Model::ScheduleRuleset.new(openstudio_model)
+        openstudio_schedule_ruleset.setName(@hash[:name])
+        schedule_type_limits = OpenStudio::Model::ScheduleTypeLimits.new(openstudio_model)
+        schedule_type_limits.setUnitType(@hash[:schedule_type_limits][:unit_type])
+        schedule_type_limits.setLowerLimitValue(@hash[:schedule_type_limits][:lower_limit_value])
+        schedule_type_limits.setUpperLimitValue(@hash[:schedule_type_limits][:upper_limit_value])
+        case 
+        when @hash[:schedule_type_limits][:numeric_type][:type] == 'ScheduleContinuous'
+          schedule_type_limits.setNumericType('Continuous')
+        when @hash[:schedule_type_limits][:numeric_type][:type] == 'ScheduleDiscrete'
+          schedule_type_limits.setNumericType('Discrete')
+        end
+        openstudio_summer_day = OpenStudio::Model::ScheduleDay.new(openstudio_model)
+        openstudio_schedule_ruleset.setSummerDesignDaySchedule(openstudio_summer_day)
+        openstudio_schedule_ruleset.summerDesignDaySchedule.setName(@hash[:summer_designday_schedule][:name])
+        if @hash[:summer_designday_schedule][:interpolate_to_timestep]
+          openstudio_schedule_ruleset.summerDesignDaySchedule.setInterpolatetoTimestep(@hash[:summer_designday_schedule][:interpolate_to_timestep])
+        else 
+          openstudio_schedule_ruleset.summerDesignDaySchedule.setInterpolatetoTimestep(@@schema[:definitions][:ScheduleDay][:properties][:interpolate_to_timestep][:default])
+        end
+        @hash[:summer_designday_schedule][:day_values].each do |day_values|
+          openstudio_schedule_ruleset.summerDesignDaySchedule.addValue(OpenStudio::Time.new(0, day_values[:time][:hour], day_values[:time][:minute],0),day_values[:value_until_time])
```

Comment:
Need to make sure that midnight is handled correctly, you will want a test or two for that

Reply:
Also, what if last time is 10pm or something before midnight, what value is used for rest of the day?

Reply:
Should it raise an error if the last day_values[:time][:hour] is not equal to 24? Since OS also requires values to be specified for a 24-hour period.